service/turn.rest/src/main/java/org/kaazing/gateway/service/turn/rest/TurnRestServiceHandler.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.service.turn.rest;
+
+import java.nio.charset.StandardCharsets;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.util.Arrays;
+
+import javax.security.auth.Subject;
+
+import org.kaazing.gateway.service.ServiceProperties;
+import org.kaazing.gateway.transport.IoHandlerAdapter;
+import org.kaazing.gateway.transport.http.HttpAcceptSession;
+import org.kaazing.gateway.transport.http.HttpHeaders;
+import org.kaazing.gateway.transport.http.HttpMethod;
+import org.kaazing.gateway.transport.http.HttpStatus;
+import org.kaazing.gateway.transport.http.HttpVersion;
+import org.kaazing.mina.core.buffer.IoBufferAllocatorEx;
+import org.kaazing.mina.core.buffer.IoBufferEx;
+
+class TurnRestServiceHandler extends IoHandlerAdapter<HttpAcceptSession> {
+    
+    private ServiceProperties options;
+    private KeyStore keystore;
+    private TurnRestCredentialsGenerator credentialGenerator;
+    private String uris;
+
+    TurnRestServiceHandler(ServiceProperties options, KeyStore keystore, TurnRestCredentialsGenerator credentialGenerator, String uris) {
+        this.options = options;
+        this.keystore = keystore;
+        this.credentialGenerator = credentialGenerator;
+        this.uris = uris;
+    }
+
+    @Override
+    protected void doSessionOpened(HttpAcceptSession session) throws Exception {        
+        HttpMethod method = session.getMethod();
+        String service = session.getParameter("service");
+      
+        String ttl = options.get("credentials.ttl");
+        Certificate alias = keystore.getCertificate(options.get("key.alias"));
+        char separator = options.get("username.separator").charAt(0);
+         
+        if (method != HttpMethod.GET) {
```

Comment:
According to https://tools.ietf.org/html/draft-uberti-rtcweb-turn-rest-00 , the method used should be POST and not GET. 

> To retrieve a new set of credentials, the client makes a HTTP POST request, specifying TURN as the service to allocate credentials for, and optionally specifying a user id parameter.